### PR TITLE
feat(serve-static): support range requests

### PR DESCRIPTION
**Pull request by yusukebe**
This PR will support [HTTP range requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests) in Serve Static Middleware. Based on #18

### CODE REVIEW

**Comment by yusukebe** [comment]
@Hoodgail 

I've created this PR, based on your PR #18. I've added some tests and additional logic. I'll merge this, attributing you as a co-author.

@tangye1234 

Could you please review this?

File: src/serve-static.ts [hunk 1]
```diff
@@ -14,36 +13,85 @@ export type ServeStaticOptions = {
   rewriteRequestPath?: (path: string) => string
 }
 
-export const serveStatic = (options: ServeStaticOptions = { root: '' }) => {
-  return async (c: Context, next: Next): Promise<Response | undefined> => {
+const createStreamBody = (stream: ReadStream) => {
```
Comment (tangye1234) [inline src/serve-static.ts:16]: Here we can use `Readable.toWeb(ReadStream)` instead, thus we can fully support backpressure.

Comment (yusukebe) [reply src/serve-static.ts]: Okay. Now let us you `Readable.toWeb()`. Thanks!

**Comment by yusukebe** [comment]
@tangye1234 

Updated!

**Review by tangye1234 — APPROVED** [review]

**Comment by yusukebe** [comment]
Thanks! Merge now.

**Pull request merged**